editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/restconfiguration/RestConfigEditor.java

```diff
@@ -0,0 +1,721 @@
+import org.eclipse.ui.forms.widgets.ScrolledForm;
+import org.eclipse.ui.forms.widgets.Section;
+import org.eclipse.ui.part.EditorPart;
+import org.fusesource.ide.camel.editor.CamelEditor;
+import org.fusesource.ide.camel.editor.internal.CamelEditorUIActivator;
+import org.fusesource.ide.camel.editor.internal.UIMessages;
+import org.fusesource.ide.camel.model.service.core.model.CamelBasicModelElement;
+import org.fusesource.ide.camel.model.service.core.model.CamelFile;
+import org.fusesource.ide.camel.model.service.core.model.ICamelModelListener;
+import org.fusesource.ide.foundation.core.util.Strings;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+/**
+ * @author bfitzpat
+ */
+public class RestConfigEditor extends EditorPart implements ICamelModelListener, ISelectionProvider {
+
+	private HashMap<String, Color> colorMap;
+	private CamelEditor parentEditor;
+	private Composite parent;
+	private ScrolledForm form;
+	private FormToolkit toolkit;
+	private HashMap<String, ArrayList<Object>> model;
+	private ImageRegistry mImageRegistry;	
+	private ListenerList<ISelectionChangedListener> listeners = new ListenerList<>();
+	private Object selection;
+	private Control selectedControl;
+	private CTabFolder tabFolder;
+	private CTabItem configurationTab;
+	private CTabItem operationsTab;
+	private Combo componentCombo;
+	private Text contextPathText;
+	private Text portText;
+	private Combo bindingModeCombo;
+	private Text hostText;
+	private Composite restOpsSection;
+	private ListViewer restList;
+
+	/**
+	 *
+	 * @param parentEditor
+	 */
+	public RestConfigEditor(CamelEditor parentEditor) {
+		this.parentEditor = parentEditor;
+	}
+
+	@Override
+	public void doSave(IProgressMonitor monitor) {
+		this.parentEditor.doSave(monitor);
+	}
+
+	@Override
+	public void doSaveAs() {
+		this.parentEditor.doSaveAs();
+	}
+
+	@Override
+	public void init(IEditorSite editorSite, IEditorInput input)
+			throws PartInitException {
+		setSite(editorSite);
+		setInput(input);
+		getSite().setSelectionProvider(this);
+	}
+
+	@Override
+	public boolean isDirty() {
+		return parentEditor.isDirty();
+	}
+
+	@Override
+	public boolean isSaveAsAllowed() {
+		return false;
+	}
+
+	private void createTabs(Composite parent) {
+		tabFolder = new CTabFolder(parent, SWT.BORDER | SWT.FLAT | SWT.RIGHT);
+		tabFolder.setSimple(false);
+		tabFolder.setLayout(new GridLayout(2, false));
+		tabFolder.setLayoutData(new GridData(GridData.FILL, GridData.FILL, true, true, 1, 10));
+		configurationTab = new CTabItem(tabFolder, SWT.NONE);
+		configurationTab.setText(UIMessages.RestConfigEditorConfigurationTab);
+
+		operationsTab = new CTabItem(tabFolder, SWT.NONE);
+		operationsTab.setText(UIMessages.RestConfigEditorOperationsTab);
+		
+		setSelectionBackground();
+	}
+
+	private void createColors() {
+		colorMap = new HashMap<>();
+		colorMap.put(RestConfigConstants.REST_COLOR_LIGHT_BLUE, new Color(Display.getDefault(), 235, 242, 250));
```

Comment:
Do you have an example of what this might look like? I like the idea of abstracting it further.

Reply:
see https://github.com/bfitzpat/fuseide/pull/14
It allows to instantiate a single time each color and don't create/dispose on each editor opening. I refactored in a second commit to use maps instead of if/then/else structure